app.js

```diff
@@ -6,12 +6,18 @@ const {getArticleById} = require ('./controllers/articles-controllers')
 
 
 
-
-
 app.use('/api', endpointsRouter);
 
 app.get('/api/articles/:article_id', getArticleById)
 
+app.use((err, req, res, next) => {
+    if (err.status && err.msg) {
+      res.status(err.status).send({ msg: err.msg });
+    } else {
+      res.status(400).send({ msg: 'Bad Request' });
```

Comment:
we don't want to have this sort of logic in this error handler if there is a specific situation where you want to send a 400 status and the msg bad request it should come from a promise reject and then it will get handled by this error handler and in your else statement you want to invoke next to pass onto the next error handler.

Also make sure your error handlers are towards the bottom of your app.js

Reply:
I don't get what exactly to do